### PR TITLE
Setup up docker, gcloud, kubectl, conda, conda envs, gcloud app creds

### PR DESCRIPTION
**Pull request by danking**
cc @tpoterba 

I'm trying to automate environment set up for new developers as much as possible.

### CODE REVIEW

**Comment by danking** [comment]
@chrisvittal can you take a look at the linux installers? installing docker on linux seems really involved so I left that out.

**Comment by danking** [comment]
I used an earlier version of this to get Alyssa set up with k8s. I think this one script more or less gets you from nothing to `kubectl get pods`.

File: env-setup.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            apt-get install gcloud || yum install gcloud
+        }
+        ;;
+    *)
+        echo "unsupported platform $PLATFORM if you think this is the wrong platform, explicitly set $HAIL_PLATFORM"
+        ;;
+esac
+
+docker version || install-docker
+conda -V || install-conda
```
[the remark this answers is not in the document]
Comment (danking) [reply env-setup.sh]: Do you always explicitly specify the path to conda when you create or update environments?

File: env-setup.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        install-docker() {
+            echo "installing docker on $PLATFORM is unsupported, please manually install: https://docs.docker.com/install/linux/docker-ce/ubuntu"
+        }
+        install-conda() {
+            tmpfile=$(mktemp /tmp/abc-script.XXXXXX)
+            curl -L https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh > $tmpfile
+            bash $tmpfile
+        }
+        install-gcloud() {
+            apt-get install gcloud || yum install gcloud
```
[the remark this answers is not in the document]
Comment (danking) [reply env-setup.sh]: I made it print an error message just like the docker install.

File: env-setup.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+PLATFORM="${HAIL_PLATFORM:-${OSTYPE}}"
+
+case "$PLATFORM" in
+    darwin*)
+        install-docker() {
+            brew cask install docker
+            open /Applications/Docker.app  # opening Docker.app seems necessary to put docker on $PATH
+        }
+        install-conda() {
+            tmpfile=$(mktemp /tmp/abc-script.XXXXXX)
```
[the remark this answers is not in the document]
Comment (danking) [reply env-setup.sh]: I switched to `$(mktemp)`.

File: env-setup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisvittal) [reply env-setup.sh]: I do some environment management with an an env var `CONDA_HOME` and an alias `Conda` which is set to `$CONDA_HOME/bin/conda` (the exact string, zsh does alias expansion and shell parameter expansion in such a way that it works out).

If I'm working in a conda env then everything just works as well and I can update envs and then reload them if necessary.

**Comment by danking** [comment]
@chrisvittal I added `$CONDA_BINARY`, which defaults to `conda`, that you can set in your `.zshrc`. Otherwise I think this is a meaningful improvement to our on-boarding experience. @cseed approve?

**Comment by chrisvittal** [comment]
@danking Thank you. This looks good for now. I'll prepare a follow on for linux installation.

**Comment by danking** [comment]
@jigold I randomly reassigned because cotton is busy

**Comment by danking** [comment]
roll again!

**Pull request merged**